src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -156,31 +156,31 @@ NamesAndTypesList ORCSchemaReader::readSchema()
         return getNamesAndRecursivelyNullableTypes(header);
     return header.getNamesAndTypesList();}
 
-void registerInputFormatORC(FormatFactory & factory)
+void registerInputFormatDeprecatedORC(FormatFactory & factory)
 {
     factory.registerInputFormat(
-            "ORC",
+            "DeprecatedORC",
```

Comment:
I think it would be more useful if the implementation was chosen based on a setting rather than format name. Similar to `output_format_parquet_use_custom_encoder`. That makes it easier to quickly disable the new implementation if a bug is found, both by the administrator (by changing server settings) and by the user (by adding a `set input_format_orc_use_fast_decoder = 0` instead of updating all the queries). Or does your approach have other advantages?

This seems easy to implement: have one `registerInputFormat()` with creator function that looks at FormatSettings and creates either ORCBlockInputFormat or DeprecatedORCBlockInputFormat.